src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.iOS.cs

```diff
@@ -1,119 +1,101 @@
-﻿#if XAMARIN_IOS
-
+﻿
 using Uno.UI.Common;
 using Windows.UI.Xaml.Data;
 
 namespace Windows.UI.Xaml.Controls
 {
 	public partial class TimePickerFlyout : Flyout
 	{
+		internal protected TimePickerSelector _timeSelector;
+		internal protected TimePickerFlyoutPresenter _timePickerPresenter;
+		internal protected FrameworkElement _headerUntapZone;
+
 		public TimePickerFlyout()
 		{
-			InitializeContent();
-		}
+			_timeSelector = new TimePickerSelector()
+			{
+				BorderThickness = Thickness.Empty,
+				HorizontalAlignment = HorizontalAlignment.Stretch,
+				HorizontalContentAlignment = HorizontalAlignment.Stretch
+			};
 
-		private void OnTap(object sender, Input.PointerRoutedEventArgs e) => e.Handled = true;
+			Content = _timeSelector;
 
-		protected internal override void Close()
+			this.Binding(nameof(Time), nameof(Time), Content, BindingMode.TwoWay);
+			this.Binding(nameof(ClockIdentifier), nameof(ClockIdentifier), Content, BindingMode.TwoWay);
+		}
+
+		protected override Control CreatePresenter()
 		{
-			var timePickerPresenter = _presenter as TimePickerFlyoutPresenter;
-			var headerUntapZone = timePickerPresenter?.FindName("HeaderUntapableZone") as FrameworkElement;
+			_timePickerPresenter = new TimePickerFlyoutPresenter() { Content = Content };
 
-			if (headerUntapZone != null)
+			void onLoad(object sender, RoutedEventArgs e)
 			{
-				headerUntapZone.PointerPressed -= OnTap;
-			}
+				_headerUntapZone = _timePickerPresenter?.FindName("HeaderUntapableZone") as FrameworkElement;
 
-			var timeSelector = Content as TimePickerSelector;
+				AttachAcceptCommand(_timePickerPresenter);
+				AttachDismissCommand(_timePickerPresenter);
 
-			if (timeSelector != null)
+				if (_timePickerPresenter != null)
+				{
+					_timePickerPresenter.Loaded -= onLoad;
+				}
+			}
+
+			if (_timePickerPresenter != null)
 			{
-				timeSelector.Cancel();
+				_timePickerPresenter.Loaded += onLoad;
```

Comment:
This will leak, use the SerialDisposable model.

Reply:
@jeromelaban this is what you meant ? The serialDisposable is set to null in onUnload